arm/authorization/client.go

```diff
@@ -0,0 +1,759 @@
+	if err != nil {
+		return result, fmt.Errorf("authorization: Failure preparing ManagementLocksClient.CreateOrUpdateAtResourceLevel request (%v)", err)
+	}
+
+	req, err = autorest.Prepare(
+		req,
+		client.WithAuthorization(),
+		client.WithInspection())
+	if err != nil {
+		return result, fmt.Errorf("authorization: Failure sending ManagementLocksClient.CreateOrUpdateAtResourceLevel request (%v)", err)
+	}
+
+	resp, err := autorest.SendWithSender(client, req)
+	if err != nil {
+		return result, fmt.Errorf("authorization: Failure sending ManagementLocksClient.CreateOrUpdateAtResourceLevel request (%v)", err)
+	}
+
+	result = ManagementLock{}
+	err = autorest.Respond(
+		resp,
+		client.ByInspecting(),
+		autorest.WithErrorUnlessOK(),
+		autorest.ByUnmarshallingJSON(&result),
+		autorest.ByClosing())
+
+	return result, err
+}
+
+func (client *ManagementLocksClient) NewCreateOrUpdateAtResourceLevelRequest(resourceGroupName string, resourceProviderNamespace string, parentResourcePath string, resourceType string, resourceName string, lockName string, parameters ManagementLock) (*http.Request, error) {
+
+	pathParameters := map[string]interface{}{
+		"lockName":                  url.QueryEscape(lockName),
+		"parentResourcePath":        parentResourcePath,
+		"resourceGroupName":         url.QueryEscape(resourceGroupName),
+		"resourceName":              url.QueryEscape(resourceName),
+		"resourceProviderNamespace": url.QueryEscape(resourceProviderNamespace),
+		"resourceType":              resourceType,
+		"subscriptionId":            url.QueryEscape(client.SubscriptionId),
+	}
+
+	queryParameters := map[string]interface{}{
+		"api-version": ApiVersion,
+	}
+
+	return autorest.DecoratePreparer(
+		client.CreateOrUpdateAtResourceLevelRequestPreparer(),
+		autorest.WithJSON(parameters),
+		autorest.WithPathParameters(pathParameters),
+		autorest.WithQueryParameters(queryParameters)).Prepare(&http.Request{})
+}
+
+func (client *ManagementLocksClient) CreateOrUpdateAtResourceLevelRequestPreparer() autorest.Preparer {
+	return autorest.CreatePreparer(
+		autorest.AsJSON(),
+		autorest.AsPut(),
+		autorest.WithBaseURL(client.BaseUri),
+		autorest.WithPath("/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{parentResourcePath}/{resourceType}/{resourceName}/providers/Microsoft.Authorization/locks/{lockName}"))
+}
+
+// DeleteAtResourceLevel deletes the management lock of a resource or any
+// level below resource.;
+//
+// resourceGroupName is the name of the resource group.
+// resourceProviderNamespace is resource identity. parentResourcePath is
+// resource identity. resourceType is resource identity. resourceName is
+// resource identity. lockName is the name of lock.
+func (client *ManagementLocksClient) DeleteAtResourceLevel(resourceGroupName string, resourceProviderNamespace string, parentResourcePath string, resourceType string, resourceName string, lockName string) (err error) {
```

Comment:
This comment is going to be about named return values, in this case `(err error)`.

Go stdlib uses named returns for documentation purposes, for instance when you return `(bool,bool,string)` it's totally ambiguous, that's why people document it as `(found bool, allowed bool, value string)` ...

I don't think named returns make sense here in this case where `error` is a well known obvious return type. Is it possible to change `(err error)` to `error` in this case.